File: CS/CustomLabel/CustomLabel.cs
using System;
using System.Drawing;
using System.ComponentModel;
using DevExpress.XtraReports;
using DevExpress.Utils.Design;
using DevExpress.XtraReports.UI;
using DevExpress.XtraReports.Localization;
using DevExpress.XtraReports.Design;
using System.ComponentModel.Design;

namespace CustomLabel {
    [
        ToolboxItem(true),
        ToolboxBitmap(typeof(MyCustomLabel), "CustomLabel16.bmp"),
        ToolboxBitmap24("CustomLabel.CustomLabel24.bmp, CustomLabel"),
        ToolboxBitmap32("CustomLabel.CustomLabel32.bmp, CustomLabel"),
        ToolboxSvgImage("CustomLabel.CustomLabel.svg, CustomLabel"),
        Designer("CustomLabel.MyCustomLabelDesigner"),
        XRDesigner("CustomLabel.MyCustomLabelDesigner"),
    ]
    public class MyCustomLabel : XRLabel {
        bool _checkValue;
        [DefaultValue(true)]
        public virtual bool CheckValue {
            get { return _checkValue; }
            set {
                if(_checkValue != value) {
                    _checkValue = value;
                    OnPropertiesChanged();
                }
            }
        }

        [Browsable(false), EditorBrowsable(EditorBrowsableState.Never), Bindable(false)]
        public override Color ForeColor { 
            get {
                return base.ForeColor;
            }
            set {
                base.ForeColor = value;
            }
        }

        public override string Text {
            get {
                return base.Text;
            }
            set {
                base.Text = value;
                OnPropertiesChanged();
            }
        }

        public MyCustomLabel() {
            this.Font = new Font("Calibri", 12, FontStyle.Bold);
            this.CheckValue = true;
        }

        private void OnPropertiesChanged() {
            int result;
            if(_checkValue)
                ForeColor = int.TryParse(Text, out result) ? Color.DarkGreen : Color.Red;
            else
                ForeColor = Color.Black;
        }
    }

    public class MyCustomLabelDesigner : XRTextControlDesigner {
        protected override void RegisterActionLists(System.ComponentModel.Design.DesignerActionListCollection list) {
            base.RegisterActionLists(list);
            list.Add(new MyCustomLabelActionList(this));
        }
    }

    public class MyCustomLabelActionList : XRComponentDesignerActionList {
        public MyCustomLabelActionList(XRComponentDesigner componentDesigner)
            : base(componentDesigner) { }

        protected override void FillActionItemCollection(DesignerActionItemCollection actionItems) {
            base.FillActionItemCollection(actionItems);
            AddPropertyItem(actionItems, "CheckValue", "CheckValue");
        }

        public bool CheckValue {
            get { return ((MyCustomLabel)this.Component).CheckValue; }
            set { this.SetPropertyValue("CheckValue", value); }
        }
    }
}

File: VB/CustomLabel/CustomLabel.vb
Imports System
Imports System.Drawing
Imports System.ComponentModel
Imports DevExpress.XtraReports
Imports DevExpress.Utils.Design
Imports DevExpress.XtraReports.UI
Imports DevExpress.XtraReports.Localization
Imports DevExpress.XtraReports.Design
Imports System.ComponentModel.Design

Namespace CustomLabel
    <
    ToolboxItem(True),
    ToolboxBitmap(GetType(CustomLabel.MyCustomLabel), "CustomLabel16.bmp"),
    ToolboxBitmap24("CustomLabel.CustomLabel24.bmp, CustomLabel"),
    ToolboxBitmap32("CustomLabel.CustomLabel32.bmp, CustomLabel"),
    ToolboxSvgImage("CustomLabel.svg, CustomLabel"),
    Designer("CustomLabel.MyCustomLabelDesigner"),
    XRDesigner("CustomLabel.MyCustomLabelDesigner")>
    Public Class MyCustomLabel
        Inherits XRLabel

        Private _checkValue As Boolean
        <DefaultValue(True)>
        Public Overridable Property CheckValue() As Boolean
            Get
                Return _checkValue
            End Get
            Set(ByVal value As Boolean)
                If _checkValue <> value Then
                    _checkValue = value
                    OnPropertiesChanged()
                End If
            End Set
        End Property

        <Browsable(False), EditorBrowsable(EditorBrowsableState.Never), Bindable(False)>
        Public Overrides Property ForeColor() As Color
            Get
                Return MyBase.ForeColor
            End Get
            Set(ByVal value As Color)
                MyBase.ForeColor = value
            End Set
        End Property

        Public Overrides Property Text() As String
            Get
                Return MyBase.Text
            End Get
            Set(ByVal value As String)
                MyBase.Text = value
                OnPropertiesChanged()
            End Set
        End Property

        Public Sub New()
            Me.Font = New Font("Calibri", 12, FontStyle.Bold)
            Me.CheckValue = True
        End Sub

        Private Sub OnPropertiesChanged()
            Dim result As Integer
            If _checkValue Then
                ForeColor = If(Integer.TryParse(Text, result), Color.DarkGreen, Color.Red)
            Else
                ForeColor = Color.Black
            End If
        End Sub
    End Class

	Public Class MyCustomLabelDesigner
		Inherits XRTextControlDesigner

		Protected Overrides Sub RegisterActionLists(ByVal list As System.ComponentModel.Design.DesignerActionListCollection)
			MyBase.RegisterActionLists(list)
			list.Add(New MyCustomLabelActionList(Me))
		End Sub
	End Class

	Public Class MyCustomLabelActionList
		Inherits XRComponentDesignerActionList

		Public Sub New(ByVal componentDesigner As XRComponentDesigner)
			MyBase.New(componentDesigner)
		End Sub

		Protected Overrides Sub FillActionItemCollection(ByVal actionItems As DesignerActionItemCollection)
			MyBase.FillActionItemCollection(actionItems)
			AddPropertyItem(actionItems, "CheckValue", "CheckValue")
		End Sub

		Public Property CheckValue() As Boolean
			Get
				Return DirectCast(Me.Component, MyCustomLabel).CheckValue
			End Get
			Set(ByVal value As Boolean)
				Me.SetPropertyValue("CheckValue", value)
			End Set
		End Property
	End Class
End Namespace
